StackTracerCore/Debugger.cs

```diff
@@ -46,175 +48,176 @@ private static void TraceWarning(string message, Exception ex)
                 SiteName = SiteName
             };
 
-            try
-            {
-                DateTime startTime = DateTime.UtcNow;
-                TraceLine("CollectTraces Function started");
+            Stopwatch sw = new Stopwatch();
+            sw.Start();
 
-                DataTarget target = null;
-                DateTime processAttachedTime = DateTime.MinValue;
+            TraceLine("CollectTraces Function started");
+            DataTarget dt = CreateDataTarget(processId);
+            stats.TimeProcessAttached = sw.ElapsedMilliseconds;
+            sw.Restart();
 
-                try
+            try
+            {
+                foreach (ClrInfo clrInfo in dt?.ClrVersions)
                 {
-                    target = DataTarget.AttachToProcess(processId, suspend: true);
-                    TraceLine($"Attached to process {processId}");
-                    processAttachedTime = DateTime.UtcNow;
-                    stats.TimeProcessAttached = DateTime.UtcNow.Subtract(startTime).TotalMilliseconds;
+                    int excludedThreads = 0;
+                    TraceLine($"Found CLR Version: {clrInfo.Version} DacFile: {clrInfo.DacInfo.PlatformSpecificFileName}, Path: {clrInfo.DacInfo.LocalDacPath}");
 
-                    foreach (ClrInfo version in target.ClrVersions)
+                    try
                     {
-                        TraceLine("Found CLR Version:" + version.Version.ToString());
-
-                        DacInfo dacInfo = version.DacInfo;
-                        TraceLine(string.Format("Dac File:  {0}", dacInfo.PlatformSpecificFileName));
-
-                        string dacLocation = version.DacInfo.LocalDacPath;
-                        if (!string.IsNullOrEmpty(dacLocation))
-                        {
-                            TraceLine("Local dac location: " + dacLocation);
-                        }
-
-                        ClrRuntime runtime = null;
-
-                        try
-                        {
-                            runtime = version.CreateRuntime(dacLocation, true);
-                        }
-                        catch (Exception ex)
-                        {
-                            TraceWarning($"StackTracer: Failed to create runtime", ex);
-                        }
-
-                        if (runtime == null)
+                        using (ClrRuntime runtime = CreateRuntime(clrInfo, clrInfo.DacInfo.LocalDacPath))
                         {
-                            continue;
-                        }
+                            if (runtime is null)
+                                continue;
 
-                        TraceLine($"ClrRuntime object created successfully for {version.Version}");
+                            var threads = runtime.Threads.Where(t => t.IsAlive);
+                            TraceLine($"Alive Thread Count = {threads.Count()}");
+                            stats.ThreadCount = threads.Count();
```

Comment:
Minor nitpick:  Using `threads` multiple times will re-enumerate the `threads` enumerable.  So every call to `.Count` below and using `threads.Take` will cause us to re-enumerate threads and filter by `.IsAlive`.  You can avoid this by _either_ changing the enumerable to an array, then using .Length...e.g.:

````C#
                            var threads = runtime.Threads.Where(t => t.IsAlive).ToArray();
                            TraceLine($"Alive Thread Count = {threads.Length}");
                            stats.ThreadCount = threads.Length;
```

Or by introducing a local variable for the count:

```C#
                            var threads = runtime.Threads.Where(t => t.IsAlive);
                            int threadCount = threads.Count();
                            TraceLine($"Alive Thread Count = {threadCount}");
                            stats.ThreadCount = threadCount;
```

The second example will still enumerate the `threads` again below using threads.Take, but it avoids an array allocation.

Reply:
Introduced a variable for count. Merging this PR now!